maven-plugin/src/test/java/org/openmrs/maven/plugins/AbstractMavenIT.java

```diff
@@ -0,0 +1,55 @@
+package org.openmrs.maven.plugins;
+
+
+import lombok.Getter;
+import lombok.Setter;
+import org.junit.Before;
+import org.openmrs.maven.plugins.utility.MavenEnvironment;
+
+import java.io.File;
+
+@Getter @Setter
+public abstract class AbstractMavenIT extends AbstractSdkIT {
+
+	MavenEnvironment mavenEnvironment = null;
+
+	@Override
+	@Before
+	public void setup() throws Exception {
+		super.setup();
+		mavenEnvironment = null;
+	}
+
+	@Override
+	void addTestResources() throws Exception {
+		includePomFile("invokeIT", "pom.xml");
+	}
+
+	protected void executeTest(MavenTestFunction testFunction) throws Exception {
+		StackTraceElement invoker = Thread.currentThread().getStackTrace()[2];
+		String className = invoker.getClassName();
+		String testMethod = invoker.getMethodName();
+		if (mavenEnvironment == null) {
+			addTaskParam("className", className);
+			addTaskParam("methodName", testMethod);
+			addTaskParam(BATCH_ANSWERS, getAnswers());
+			addTaskParam("testMode", "true");
+			String plugin = resolveSdkArtifact();
+			verifier.executeGoal(plugin + ":" + InvokeMethod.NAME);
+		}
+		else {
+			testFunction.executeTest();
+		}
```

Comment:
So the general idea here, is that a particular integration test will wrap it's test functionality in this `executeTest` method.  When this is run, the mavenEnvironment will be null, and this will trigger the new "InvokeMethod" plugin to run it, by creating a new instance of this same test class, and executing this same method, but this time with the mavenEnvironment populated from the InvokeMethod plugin.